Additions/ImageLoader.cs

```diff
@@ -98,6 +98,10 @@ private static async Task<Bitmap> InnerLoadImageAsync(CancellationToken ct, stri
 		{
 			TaskCompletionSource<Android.Graphics.Bitmap> source = new TaskCompletionSource<Android.Graphics.Bitmap>();
 
+			//Propagate cancellation to the managed TCS, even though we don't propagate it to Universal Image Loader. This is because UIL has 
+			// the air of cancelling downloads itself if another uri is requested for the same ImageView, leading us to wait on a task that 
+			// will never complete (thanks to the concurrency logic reusing existing tasks).
+			ct.Register(() => source.TrySetCanceled()); 

```

Comment:
I think `ct.Register` would create a disposable that will never be disposed?

Reply:
Fair point. Wrapped it in a using, since we already await the TCS within the same method.